src/pages/new-chat/new-chat.html

```diff
@@ -10,9 +10,15 @@
 
 <ion-content class="new-chat">
   <ion-list class="users">
-    <button ion-item *ngFor="let user of users | async" class="user" (click)="addChat(user)">
+    <button ion-item *ngFor="let user of users | async | search:searchPattern:'profile.name'" class="user" (click)="addChat(user)">
```

Comment:
Your pipe comes after the `async` which means that it will get the array value from the subscription. 
Maybe put it before `async` and then use `filter` operator of RxJS on your stream, instead `filter` of Array?